### ISSUE TITLE
Can't use azureFile in Germany Cloud Central

### ISSUE THREAD

**Issue by Sennahoi** [Azure/acs-engine#2400]
**Is this a request for help?**:
Maybe.
---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
ISSUE.
---

**What version of acs-engine?**:
0.13.1
---

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
Kubernetes 1.7.9

**What happened**:

Using the following template

```
{
  "apiVersion": "vlabs",
  "location": "germanycentral",
  "properties": {
    "orchestratorProfile": {
      "orchestratorType": "Kubernetes"
    },
    "masterProfile": {
      "count": 1,
      "dnsPrefix": "kubernetesmaster",
      "vmSize": "Standard_D2_v2"
    },
    "agentPoolProfiles": [
      {
        "name": "agentpool1",
        "count": 3,
        "vmSize": "Standard_D2_v2",
        "availabilityProfile": "AvailabilitySet"
      }
    ],
    "linuxProfile": {
      "adminUsername": "xxx",
      "ssh": {
        "publicKeys": [
          {
            "keyData": "ssh-rsa xxx"
          }
        ]
      }
    },
    "servicePrincipalProfile": {
      "clientId": "xxx",
      "secret": "xxx"
    }
  }
}
```

I try to setup a pod with a storage account with this file
```
apiVersion: v1
kind: Pod
metadata:
 name: storagetest
spec:
 containers:
  - image: nginx
    name: nginx
    volumeMounts:
      - name: azure
        mountPath: /mnt/azure
 volumes:
      - name: azure
        azureFile:
          secretName: azure-secret
          shareName: clusterstorage
          readOnly: false
```

When I start the pod, I get the following error message
```
Mounting command: mount
Mounting arguments: //clusterstorage.file.core.windows.net/clusterstorage /var/lib/kubelet/pods/xxx/volumes/kubernetes.io~azure-file/azure cifs [vers=3.0,username=clusterstorage,password=xxx/+xxx/JUX/xxx==,dir_mode=0777,file_mode=0777]
Output: mount error(13): Permission denied
Refer to the mount.cifs(8) manual page (e.g. man mount.cifs)
```

From the Azure Portal, I know that the url ends with blob.core.cloudapi.de. So, I guess that may be the problem here. 

**What you expected to happen**:
That the pod starts and the volume will be mounted.

**How to reproduce it** (as minimally and precisely as possible):
As described above.

**Comment by jackfrancis** [comment]
@Sennahoi Are you able to confirm @karataliu's fix by building a 1.7.13 cluster?

**Comment by Sennahoi** [comment]
Sorry for not responding. I switched to 1.8.x and the problem was gone. Thanks for your help!
